src/pages/Users/Users.jsx

```diff
@@ -0,0 +1,368 @@
+    const formatedData = {
+      ...data,
+      dateOfBirth: new Date(dateOfBirth).getTime(),
+      startDate: new Date(startDate).getTime(),
+    };
+
+    const userId = await addUserData(editUserId, formatedData);
+    if (!editUserId) {
+      sendActivateLinkToEmail(data.email, userId);
+    }
+    this.setState({
+      data: null,
+    });
+    this.getData({
+      editUserId: null,
+      editUserData: null,
+    });
+
+    this.closeModal();
+  }
+
+  async editUser(userId) {
+    const { data } = this.state;
+    const editUserData = data.filter((userData) => {
+      return userData.id === userId;
+    })[0];
+    this.setState({
+      editUserData,
+      editUserId: userId,
+    });
+    this.openModal();
+  }
+
+  async deleteUser(userId) {
+    await deleteUser(userId);
+    this.setState({
+      data: null,
+    });
+    this.getData();
+  }
+
+  openUserProfile(userId) {
+    const { data } = this.state;
+    const readUserData = data.filter((userData) => {
+      return userData.id === userId;
+    })[0];
+    this.setState({
+      readUserData,
+      readOnly: true,
+    });
+    this.openModal();
+  }
+
+  openModal() {
+    this.setState({
+      isModalActive: true,
+    });
+  }
+
+  closeModal() {
+    this.setState({
+      readUserData: null,
+      editUserData: null,
+      editUserId: null,
+      readOnly: false,
+      isModalActive: false,
+    });
+  }
+
+  render() {
+    const { match } = this.props;
+    const { data, isModalActive, editUserData, readUserData, readOnly } = this.state;
+    let formTitle;
+    if (editUserData) {
+      formTitle = USERS_TITLES.EDIT;
+    } else if (readUserData) {
+      formTitle = USERS_TITLES.READ;
+    } else {
+      formTitle = USERS_TITLES.CREATE;
+    }
+
+    return (
+      <main className={classes.main}>
+        <UniversalButton onClick={this.onOpenModal}>Create</UniversalButton>
+        {isModalActive && (
+          <Modal className={classes.addUserModal} onClose={this.onCloseModal}>
+            <ModalForm
+              contentClassName={classes.formContent}
+              controls={getUserFormControls()}
+              onSubmit={this.addUserHandler}
+              onCloseForm={this.onCloseModal}
+              startData={readUserData || editUserData}
+              readOnly={readOnly}
+              submitButtonText={editUserData ? USERS_BUTTONS.EDIT : USERS_BUTTONS.CREATE}
+              title={formTitle}
+            />
+          </Modal>
+        )}
+        {data && (
+          <>
+            <h2 className={classes.title}>{USERS_TITLES.PAGE}</h2>
+            <UsersGrid
+              data={data}
+              onWatchData={this.onWatchUserData}
+              onEditUser={this.editUserHandler}
+              onDeleteUser={this.deleteUserHandler}
+              path={match.path}
+            />
+          </>
+        )}
+      </main>
+    );
+  }
+}
+
+Users.propTypes = {
+  match: PropTypes.objectOf(PropTypes.any).isRequired,
```

Comment:
don't use `any`

Reply:
Now I donn't use ```match``` in this component, but I fix match propType in other places.
```suggestion
match: PropTypes.shape({
    params: PropTypes.shape({
      name: PropTypes.string.isRequired
    })
  }),
};
```